### ISSUE TITLE
How to read a FITS file with FitsReader and save it into a fitsHeaderDataUnit?Could anyone give me any sample please.

### ISSUE THREAD

**Issue by RyanTaozr** [Tigra-Astronomy/TA.ObjectOrientedAstronomy#6]
Code I've tried but failed:“ArgumentException: illegal characters in path”

            
          //Create new OpenFileDialog
           OpenFileDialog openFileDialog = new OpenFileDialog
            {
                InitialDirectory = "c:\\download",
                Filter = "(*FITS)|*FITS|All files (*.*)|*.*",
                Title = "Select .FITS File",
                RestoreDirectory = true,
            };
            
            // Show open file dialog box
            Nullable<bool> result = openFileDialog.ShowDialog();

            // Process open file dialog box results
            if (result == true)
            {
                //Get the path of specified file
                filePath = openFileDialog.FileName;
                TextBox2.Text = openFileDialog.FileName;

                FitsHeader fitsHeader;
                FitsHeaderDataUnit fitsHeaderDataUnit ;

                FileStream fs = new FileStream(openFileDialog.FileName, FileMode.Open, FileAccess.Read);
                FitsReader fitsReader = new FitsReader(fs);

                fitsHeader = fitsReader.ReadPrimaryHeader().Result;
                fitsHeaderDataUnit = fitsReader.ReadPrimaryHeaderDataUnit().Result;
             }

**Comment by NameOfTheDragon** [comment]
Have you looked at the sample FITS Viewer application? That should show you exactly how to do it. Here's an abstraction from that code...
```lang=csharp
                FitsHeaderDataUnit primaryHdu;
                using (var stream = FitsFileOpener.OpenFile())
                    {
                    var reader = new FitsReader(stream);
                    primaryHdu = await reader.ReadPrimaryHeaderDataUnit().ConfigureAwait(true);
                    }
                var image = primaryHdu.ToWindowsBitmap();
```

**Comment by NameOfTheDragon** [comment]
The sample application has been migrated to https://github.com/Tigra-Astronomy/TA.ObjectOrientedAstronomy.Samples

**Comment by NameOfTheDragon** [comment]
Closing issue because a sample application has been provided which shows exactly how to do what was requested. @RyanTaozr

**Comment by RyanTaozr** [comment]
Thank you very much for your help, the samples you provided and this project helped me to complete my graduation project for my undergraduate degree, I sincerely appreciate it！！


RyanTao
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Tigra-Astronomy/TA.ObjectOrientedAstronomy"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月21日(星期天) 凌晨5:15
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Tigra-Astronomy/TA.ObjectOrientedAstronomy] How to read a FITS file with FitsReader and save it into a fitsHeaderDataUnit?Could anyone give me any sample please. (Issue #6)





 
Closed #6 as completed.
